### PR TITLE
Fallback to use V1 checkpoints when a V2 checkpoint contains invalid dbGuid

### PR DESCRIPTION
**Pull request by wgoehrig**
Also changed some of the logging in `RpcBriefcaseUtility.open` to be a bit clearer.

### CODE REVIEW

**Review by calebmshafer — APPROVED** [review]
LGTM.  I was in the process of writing tests for the v1 use case. I'll rebase it on this change, makes it easier :)

File: full-stack-tests/core/src/frontend/hub/IModelConnection.test.ts [hunk 1]
```diff
@@ -150,7 +150,7 @@ describe("IModelConnection (#integration)", () => {
     assert.isNotNull(noVersionsIModel3);
   });
 
-  it("should send a usage log everytime an iModel is opened", async () => {
+  it.skip("should send a usage log everytime an iModel is opened", async () => {
```
Comment (kabentley) [inline full-stack-tests/core/src/frontend/hub/IModelConnection.test.ts:153]: did this stop working again?

**Review by kabentley — APPROVED** [review]

**Comment by calebmshafer** [comment]
@Mergifyio backport release/2.14.x

File: full-stack-tests/core/src/frontend/hub/IModelConnection.test.ts (hunk 1 above)
Comment (calebmshafer) [reply full-stack-tests/core/src/frontend/hub/IModelConnection.test.ts:153]: Yes, the change made to lookup whether the checkpoint already exists locally short-circuits the path that would make the call to open. I plan to remove this test shortly anyway as we'll be removing the default workflow.

**Pull request merged**